File: frontend/src/types/Person.ts
import { Student } from "./Student";

export type Person = {
  id?: number;
  firstName?: string;
  middleName?: string;
  lastName?: string;
  birthDate?: string | Date;
  sex?: boolean;
  avatarFileName?: string;
  studentEntity?: Student;
  passportEntity?: { scanFileName: string };
};

File: api/Models/Person.cs
namespace api.Model
{
    public class Person
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = null!;
        public string MiddleName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public DateTime BirthDate { get; set; } = new();
        public int Sex { get; set; }
        public string AvatarFileName { get; set; } = null!;

        public Passport? Passport { get; set; }
        public Student? Student { get; set; }
    }
}